src/mesh/voxel_cloud.rs

```diff
@@ -995,6 +1015,45 @@ impl ScalarField {
         }
     }
 
+    /// Add values of the other scalar fields to the respective voxels of the
+    /// current scalar field. The current scalar field will not be resized, so
+    /// only the overlapping areas will be will be added. If the voxel in the
+    /// current or the other scalar field is None (does not contain any voxel or
+    /// does not exist), it is treated according to the selected
+    /// `empty_voxels_treatment` - it is either replaced with a constant value
+    /// or considered None and then also the resulting voxel will become None.
+    #[allow(dead_code)]
+    pub fn add_values(&mut self, other: &ScalarField, empty_voxels_treatment: TreatEmpty) {
+        for (one_dimensional, voxel) in self.voxels.iter_mut().enumerate() {
+            let cartesian_coordinate = one_dimensional_to_cartesian_coordinate(
+                one_dimensional,
+                &self.block_start,
+                &self.block_dimensions,
+                &self.voxel_dimensions,
+            );
+            let absolute_coordinate_other = cartesian_to_absolute_voxel_coordinate(
+                &cartesian_coordinate,
+                &other.voxel_dimensions,
+            );
+            let voxel_other = other.value_at_absolute_voxel_coordinate(&absolute_coordinate_other);
+
+            match empty_voxels_treatment {
+                TreatEmpty::AsNone => {
+                    *voxel = if let (Some(value_self), Some(value_other)) = (&voxel, voxel_other) {
+                        Some(value_self + value_other)
+                    } else {
+                        None
```

Comment:
Whoah! I think this resets `voxel` to `None` even if by itself it is not `None`. Shouldn't this be `voxel`? Or even better, don't assign to it in the else branch.

Reply:
okay, here i can imagine what i couldn't imagine in the previous comment. this behavior is similar to the interpolation function